api/src/main/java/com/intern/api/client/DataBaseClient.java

```diff
@@ -49,19 +50,30 @@ public Document deleteMessage(Long messageId) {
     return collection.find(eq("from.id", userId)).limit(limitAmount).into(new ArrayList<>());
   }
 
-  public Document getUserInfo(String username) {
-    return collection.find(eq("from.username", username)).first().get("from", Document.class);
+  public Document getUserInfo(String username, Optional<Integer> charId) {
+    Document result;
+    if (charId.isPresent()) {
+      result =
+          collection
+              .find(and(eq("from.username", username), eq("chat.id", charId.get())))
+              .first()
+              .get("from", Document.class);
+    } else {
+      result = collection.find(eq("from.username", username)).first().get("from", Document.class);
+    }
```

Comment:
If you use optional, you can have this code written this way:
```
return chatId
    .map(chatId -> collection...)
    .orElse(collection...)
```
`Optional.map` won't be called when `Optional` is empty.